From 221d8e557ab5a00246f7b09746377819cfbaec5e Mon Sep 17 00:00:00 2001
From: Imran Desai <imran.desai@intel.com>
Date: Wed, 9 Mar 2022 10:24:45 -0700
Subject: [PATCH 05/17] tools/tpm2_tool.c: Fix an issue where LOG_WARN is
 always displayed

Despite setting the 'quiet' flag with -Q the warning messages were
always displayed.

Signed-off-by: Imran Desai <imran.desai@intel.com>
---
 lib/tpm2_options.c          | 12 +++++++++---
 tools/tpm2_makecredential.c |  9 ++++++---
 2 files changed, 15 insertions(+), 6 deletions(-)

diff --git a/lib/tpm2_options.c b/lib/tpm2_options.c
index 8c8af2af..1238e440 100644
--- a/lib/tpm2_options.c
+++ b/lib/tpm2_options.c
@@ -456,12 +456,16 @@ tpm2_option_code tpm2_handle_options(int argc, char **argv,
 
         /* tool doesn't request a sapi, don't initialize one */
         if (flags->tcti_none && is_optional_sapi) {
-            LOG_WARN("Tool optionally uses SAPI. Continuing with tcti=none");
+            if (!flags->quiet) {
+                LOG_WARN("Tool optionally uses SAPI. Continuing with tcti=none");
+            }
             goto none;
         }
 
         if (flags->tcti_none && is_no_sapi) {
-            LOG_WARN("Tool does not use SAPI. Continuing with tcti=none");
+            if (!flags->quiet) {
+                LOG_WARN("Tool does not use SAPI. Continuing with tcti=none");
+            }
             goto none;
         }
 
@@ -481,7 +485,9 @@ tpm2_option_code tpm2_handle_options(int argc, char **argv,
         bool is_optional_fake_tcti = (flags->tcti_none && tool_opts &&
             tool_opts->flags & TPM2_OPTIONS_OPTIONAL_SAPI_AND_FAKE_TCTI);
         if (is_optional_fake_tcti) {
-            LOG_WARN("Tool optionally uses SAPI. Continuing with tcti=fake");
+            if (!flags->quiet) {
+                LOG_WARN("Tool optionally uses SAPI. Continuing with tcti=fake");
+            }
             *tcti = (TSS2_TCTI_CONTEXT *)&fake_tcti;
             goto none;
         }
diff --git a/tools/tpm2_makecredential.c b/tools/tpm2_makecredential.c
index 0b0fa123..5bd5b484 100644
--- a/tools/tpm2_makecredential.c
+++ b/tools/tpm2_makecredential.c
@@ -310,11 +310,14 @@ static void set_default_TCG_EK_template(TPMI_ALG_PUBLIC alg) {
     ctx.public.publicArea.nameAlg = TPM2_ALG_SHA256;
 }
 
-static tool_rc process_input(void) {
+static tool_rc process_input(tpm2_option_flags flags) {
 
     TPMI_ALG_PUBLIC alg = TPM2_ALG_NULL;
     if (ctx.key_type) {
-        LOG_WARN("Because **-G** is specified, assuming input encryption public key is in PEM format.");
+        if (!flags.quiet) {
+            LOG_WARN("Because **-G** is specified, assuming input encryption "
+                     "public key is in PEM format.");
+        }
         alg = tpm2_alg_util_from_optarg(ctx.key_type,
             tpm2_alg_util_flags_asymmetric);
         if (alg == TPM2_ALG_ERROR ||
@@ -379,7 +382,7 @@ static tool_rc tpm2_tool_onrun(ESYS_CONTEXT *ectx, tpm2_option_flags flags) {
 
     UNUSED(flags);
 
-    tool_rc rc = process_input();
+    tool_rc rc = process_input(flags);
     if (rc != tool_rc_success) {
         return rc;
     }
-- 
2.40.1

